app/services/document_generator_service.py

```
@@ -0,0 +1,600 @@
+                                run.bold = True
+
+                    # Data rows
+                    for row_idx, row_data in enumerate(rows):
+                        for col_idx, cell_val in enumerate(row_data):
+                            if col_idx < num_cols:
+                                table.rows[row_idx + 1].cells[col_idx].text = str(cell_val)
+
+                    doc.add_paragraph("")  # Spacer after table
+
+            elif block.type == "code":
+                lang = block.metadata.get("language", "")
```

Ruff (0.15.1)

[error] Local variable `lang` is assigned to but never used

Remove assignment to unused variable `lang`

(F841)